src/rabbit_channel.erl

```diff
@@ -1846,30 +1783,32 @@ consumer_monitor(ConsumerTag,
     State#ch{queue_monitors  = maybe_monitor(QRef, QMons),
              queue_consumers = QCons1}.
 
-track_delivering_queue(NoAck, QPid, QName,
-                       State = #ch{queue_names = QNames,
-                                   queue_monitors = QMons,
+track_delivering_queue(NoAck, QPid,
+                       State = #ch{
+                                   % queue_monitors = QMons,
                                    delivering_queues = DQ}) ->
     QRef = qpid_to_ref(QPid),
-    State#ch{queue_names = maps:put(QRef, QName, QNames),
-             queue_monitors = maybe_monitor(QRef, QMons),
+    State#ch{
+             % queue_monitors = maybe_monitor(QRef, QMons),
              delivering_queues = case NoAck of
                                      true  -> DQ;
                                      false -> sets:add_element(QRef, DQ)
                                  end}.
 
 handle_publishing_queue_down(QPid, Reason,
                              State = #ch{unconfirmed = UC,
-                                         queue_names = QNames})
+                                         queue_states = QStates})
   when ?IS_CLASSIC(QPid) ->
```

Comment:
These macros are not generic either. Should we have a `rabbit_queue_type:needs_handle_publishing_queue_down` or something similar?

The same applies to all other ocurrences of this and the `?IS_QUORUM` macro. Now that we have types they should disappear

Reply:
I agree and to do that we'd have to move all the confirm accounting inside the queue type module. I change I found a bit scary at this point.

Reply:
So having pondered this a bit more the current scope of the queue type abstraction isn't complete, it would need to handle confirms, downs etc as well to be considered so. Taht said I think at this point it achieves one goal: it makes it easier for us to deliver new queue types. The secondary goal of completing the queue type abstraction and making it easier for us to write new channel processes is one we can choose to defer at this point.